statistics/builder.go

```diff
@@ -110,10 +110,11 @@ func BuildColumn(ctx sessionctx.Context, numBuckets, id int64, collector *Sample
 		ndv = count
 	}
 	hg := NewHistogram(id, ndv, collector.NullCount, 0, tp, int(numBuckets))
-	valuesPerBucket := float64(count)/float64(numBuckets) + 1
 
 	// As we use samples to build the histogram, the bucket number and repeat should multiply a factor.
 	sampleFactor := float64(count) / float64(len(samples))
+	// Since the bucket count is increased by sampleFactor, so add sampleFactor to avoid building too many buckets.
```

Comment:
Can not understand this.

Reply:
If the sample factor is 2, and the max value per bucket is 3, and if you have 100 samples, it would build (100/2) buckets, not (100/3) buckets, because now every bucket will have depth 2, not 3, since the bucket count is increased by 2, not 1.